OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -541,9 +538,7 @@ public void ResolveOrder(Actor self, Order order)
 					}
 					else
 					{
-						var res = order.TargetActor.TraitOrDefault<Reservable>();
-						if (res != null)
-							Reservation = res.Reserve(order.TargetActor, self, this);
+						MakeReservation(order.TargetActor);
```

Comment:
This will now mean `UnReserve` is called before setting a new reservation and is the likely fix for #9783.